terraform/charm/simple_deployment/main.tf

```diff
@@ -0,0 +1,89 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Identify if the current model is where the main orchestrator resides to deploy the tls provider
+locals {
+  is_main_orchestrator = !lookup(var.config, "init_hold", false)
+}
+
+#--------------------------------------------------------
+# 1. DEPLOYMENTS
+#--------------------------------------------------------
+
+# Deploy required applications
+resource "juju_application" "opensearch" {
+  charm {
+    name     = "opensearch"
+    channel  = var.channel
+    revision = var.revision
```

Comment:
`revision == null` works fine?

Reply:
Yes. When specifying the `channel`, the revision is optional.